tests/testthat/test-create.R

```diff
@@ -29,31 +27,59 @@ test_that("create functions return path to new proj, but restore active proj", {
 
 test_that("nested package is disallowed, by default", {
   dir <- scoped_temporary_package()
-  expect_usethis_error(scoped_temporary_package(path(dir, "abcde")), "anyway")
+  expect_usethis_error(create_package(path(dir, "abcde")), "anyway")
 })
 
 test_that("nested project is disallowed, by default", {
   dir <- scoped_temporary_project()
-  expect_usethis_error(scoped_temporary_project(path(dir, "abcde")), "anyway")
+  expect_usethis_error(create_project(path(dir, "abcde")), "anyway")
+})
+
+test_that("nested package can be created if user really, really wants to", {
+  parent <- scoped_temporary_package()
+  with_mock(
+    # since user can't approve interactively, use the backdoor
+    `usethis:::allow_nested_project` = function() TRUE,
+    child <- create_package(path(parent, "fghijk"))
+  )
+  expect_true(possibly_in_proj(child))
+  expect_true(is_package(child))
+})
+
+test_that("nested project can be created if user really, really wants to", {
+  parent <- scoped_temporary_project()
+  with_mock(
+    # since user can't approve interactively, use the backdoor
+    `usethis:::allow_nested_project` = function() TRUE,
+    child <- create_project(path(parent, "fghijk"))
+  )
+  expect_true(possibly_in_proj(child))
+  expect_false(is_package(child))
```

Comment:
Until now, we've never actually created a nested package or project in the tests.